sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/EventTracking.cs

```diff
@@ -0,0 +1,246 @@
+        var remainder = partitionCount % numPublishers;
+
+        var startPartition = 0;
+        var endPartition = 0;
+
+        if (thisPublisherIndex >= remainder)
+        {
+            startPartition = (baseNum*thisPublisherIndex) + remainder;
+            endPartition = startPartition + baseNum;
+        }
+        else
+        {
+            startPartition = (baseNum*thisPublisherIndex) + thisPublisherIndex;
+            endPartition = startPartition + baseNum + 1;
+        }
+
+        var assignedPartitions = new List<string>();
+
+        for (int i = startPartition; i < endPartition; i++)
+        {
+            assignedPartitions.Add(partitionIds[i]);
+        }
+
+        return assignedPartitions;
+    }
+
+    /// <summary>
+    ///   Validates an event. This instance checks if the event has an id, if the event has already been seen before by this instance, that the
+    ///   intended partition was the one the event was received from, that the event was received in order, and that none were missed, and that
+    ///   the body is valid.
+    /// </summary>
+    ///
+    /// <param name="eventData"></param>
+    /// <param name="partitionReceivedFrom"></param>
+    /// <param name="metrics"></param>
+    ///
+    /// <returns>The publisher assigned index number of the event that was just read, or -1 if this event was unkown or a duplicate.</returns>
+    ///
+    /// <remarks>
+    ///   One instance should be created for each processor or consumer role. This instance will be able to keep track of all the events the
+    ///   processor or consumer has received for each partition it is reading or processing from.
+    /// </remarks>
+    ///
+    private int _checkEvent(EventData eventData, string partitionReceivedFrom, Metrics metrics)
+    {
+        // Id Checks
+        var hasID = eventData.Properties.TryGetValue(IdPropertyName, out var eventIdProperty);
+        var eventId = eventIdProperty?.ToString();
+
+        if (!hasID)
+        {
+            metrics.Client.GetMetric(Metrics.UnknownEventsProcessed).TrackValue(1);
+            return -1;
+        }
+
+        if (ReadEvents.ContainsKey(eventId))
+        {
+            metrics.Client.GetMetric(Metrics.DuplicateEventsDiscarded).TrackValue(1);
+            return -1;
+        }
+
+        // Partition Checks
+        var hasPartition = eventData.Properties.TryGetValue(PartitionPropertyName, out var partitionProperty);
+        var publisherSetPartitionProperty = partitionProperty?.ToString();;
+
+        if (partitionReceivedFrom != publisherSetPartitionProperty)
+        {
+            metrics.Client.GetMetric(Metrics.EventReceivedFromWrongPartition).TrackValue(1); //want an event here? or exception? This doesn't give much info
+        }
+
+        // Index Value checks
+        var hasIndex = eventData.Properties.TryGetValue(IndexNumberPropertyName, out var IndexProperty);
+        int.TryParse(IndexProperty.ToString(), out var indexNumber);
```

Comment:
We should consider what happens if this isn't present or isn't numeric here.    That will default to 0, which may track as valid.     Likewise, if `hasIndex` is `false` this line will throw.